src/Jobs/RefreshTokenDeletionDispatcher.php

```diff
@@ -66,15 +66,22 @@ public function handle()
 
             $recipients = NotificationRecipient::all()
                 ->filter(function ($recipient) {
-                    return $recipient->shouldReceive('refresh_token');
+                    return $recipient->shouldReceive(AbstractRefreshTokenNotification::class);
                 });
 
             if($recipients->isEmpty()){
                 logger()->debug('No Receiver found for this Notification. This job is going to be deleted.');
                 $this->delete();
             }
 
-            Notification::send($recipients, (new RefreshTokenDeletedNotification($this->refresh_token)));
+            $recipients->groupBy('driver')
+                ->each(function ($grouped_recipients) {
+                    $driver = $grouped_recipients->first()->driver;
+                    $notification_class = AbstractRefreshTokenNotification::getDiversImplementations()[$driver];
```

Comment:
Maybe we can implement a `getDriverImplementation(string $driver): INotification` which will return the requested driver implementation based on `getDriversImplementations()`.

In case the requested driver does not exists, throw a `UnknownDriverException`.